File: app/src/main/java/com/helpcrunch/demo/design/CustomTheme.java
package com.helpcrunch.demo.design;

import android.graphics.Color;

import androidx.annotation.ColorRes;

import com.helpcrunch.demo.R;
import com.helpcrunch.library.core.options.design.HCAvatarTheme;
import com.helpcrunch.library.core.options.design.HCChatAreaTheme;
import com.helpcrunch.library.core.options.design.HCMessageAreaTheme;
import com.helpcrunch.library.core.options.design.HCNotificationsTheme;
import com.helpcrunch.library.core.options.design.HCPreChatTheme;
import com.helpcrunch.library.core.options.design.HCSystemAlertsTheme;
import com.helpcrunch.library.core.options.design.HCTheme;
import com.helpcrunch.library.core.options.design.HCToolbarAreaTheme;

public class CustomTheme {
    public static HCTheme getTheme() {
        @ColorRes
        int colorWhite = R.color.hc_color_white_dark;

        HCTheme.Builder builder = new HCTheme.Builder(HCTheme.Type.DARK);

        HCAvatarTheme avatarTheme = new HCAvatarTheme.Builder()
                .setPlaceholderBackgroundColor(R.color.hc_color_chat_bubble_dark_other)
                .setPlaceholderTextColor(colorWhite)
                .build();

        // It so happened historically that the theme that is applied to the chat
        // will be applied to the CV and to the chat list.
        // The text color will be matched to the background color and the avatar theme
        // will be copied from the chat avatar theme
        HCChatAreaTheme chatAreaTheme = new HCChatAreaTheme.Builder()
                .setBackgroundColor(R.color.hc_color_chat_dark_bg)
                .setAdditionalMessagesBackgroundColor(R.color.hc_color_chat_bubble_dark_other)
                .setIncomingBubbleColor(R.color.hc_color_chat_bubble_dark_other)
                .setOutcomingBubbleColor(R.color.hc_color_chat_bubble_dark_me)
                .setIncomingBubbleTextColor(colorWhite)
                .setOutcomingBubbleTextColor(colorWhite)
                .setIncomingBubbleLinkColor(colorWhite)
                .setOutcomingBubbleLinkColor(colorWhite)
                .setSystemMessageColor(colorWhite)
                .setTimeTextColor(R.color.hc_color_white_op40)
                .setProgressViewsColor(colorWhite)
                .setFabDownBackgroundColor(R.color.hc_color_toolbar_dark_bg)
                .setBrandingType(HCChatAreaTheme.Branding.DARK)
                .setAvatarTheme(avatarTheme)
                .setIncomingCodeBackgroundColor(R.color.hc_color_code_bg_incoming_dark)
                .setOutcomingCodeBackgroundColor(R.color.hc_color_code_bg_outcoming_dark)
                .setIncomingCodeTextColor(R.color.hc_color_code_color)
                .setOutcomingCodeTextColor(R.color.hc_color_code_color)
                .setIncomingBlockQuoteColor(R.color.hc_color_block_quote_incoming_dark)
                .setOutcomingBlockQuoteColor(R.color.hc_color_block_quote_outcoming_dark)
                // File
                .setIncomingFileTextColor(R.color.hc_color_white)
                .setOutcomingFileTextColor(R.color.hc_color_white)
                .setOutcomingFileBackgroundColor(R.color.hc_color_code_color)
                .setIncomingFileBackgroundColor(R.color.hc_color_code_color)
                .setOutcomingFileIconColor(R.color.hc_color_code_color)
                .setIncomingFileIconColor(R.color.hc_color_white)
                .setAttachmentIconsColor(R.color.hc_color_chat_bubble_dark_other)
                .setAuthorNameColor(colorWhite)
                .build();

        HCMessageAreaTheme messageAreaTheme = new HCMessageAreaTheme.Builder()
                .setButtonType(HCMessageAreaTheme.ButtonType.ICON)
                .setAttachmentsIcon(R.drawable.ic_hc_attach_file)
                .setButtonSendBackgroundSelector(R.drawable.bg_hc_selector_btn_send_dark)
                .setInputFieldTextColor(R.color.hc_color_chats_text_dark)
                .setInputFieldTextHintColor(R.color.hc_color_chats_text_hint_dark)
                .setInputOutlineColor(R.color.hc_color_toolbar_dark_bg)
                .setBackgroundColor(R.color.hc_color_message_bar_dark_bg)
                .setMessageMenuBackgroundColor(R.color.hc_color_chat_dark_bg)
                .setMessageMenuSummaryTextColor(R.color.hc_color_menu_summary_color_dark)
                .setMessageMenuIconColor(R.color.hc_color_menu_summary_color_dark)
                .setMessageMenuTextColor(colorWhite)
                .build();

        HCToolbarAreaTheme toolbarAreaTheme = new HCToolbarAreaTheme.Builder()
                .setBackgroundColor(R.color.hc_color_toolbar_dark_bg)
                .setAgentsTextColor(colorWhite)
                .setStatusBarColor(R.color.hc_color_toolbar_dark_bg)
                .setStatusBarLight(false)
                .setOutlineColor(R.color.hc_color_chat_dark_bg)
                .setBackButtonDrawableRes(R.drawable.ic_hc_arrow_back)
                .setAvatarTheme(avatarTheme)
                .build();

        HCSystemAlertsTheme systemAlertsTheme = new HCSystemAlertsTheme.Builder()
                .setToastsBackgroundColor(R.color.hc_color_white_op90)
                .setToastsTextColor(R.color.hc_main_dark)
                .setDialogsHeaderColor(R.color.hc_color_toolbar_dark_bg)
                .setWelcomeMessageBackgroundColor(R.color.hc_color_toolbar_dark_bg)
                .setWelcomeMessageTextColor(colorWhite)
                .setWarningDialogsHeaderColor(R.color.hc_color_bg_button_enabled_dark)
                .setDialogAcceptButtonTextColor(colorWhite)
                .setDialogCancelButtonTextColor(colorWhite)
                .setDialogBackgroundColor(R.color.hc_color_chat_dark_bg)
                .setDialogMessageTextColor(colorWhite)
                .build();

        HCPreChatTheme preChatTheme = new HCPreChatTheme.Builder()
                .setButtonContinueBackgroundSelector(R.drawable.btn_hc_rounded_dark)
                .setInputFieldTextColor(R.color.hc_color_chats_text_dark)
                .setInputFieldTextHintColor(R.color.hc_color_chats_text_hint_dark)
                .setInputFieldBackgroundDrawableRes(R.drawable.bg_hc_chat_field_dark)
                .setBackgroundColor(R.color.hc_color_chat_dark_bg)
                .setMessageBackgroundColor(R.color.hc_color_toolbar_dark_bg)
                .setMessageTextColor(colorWhite)
                .build();

        HCNotificationsTheme notificationsTheme = new HCNotificationsTheme.Builder()
                .setAvatarTheme(avatarTheme)
                .setColor(Color.parseColor("#4D4D7F")) // (sic!)
                .build();

        builder.setChatAreaTheme(chatAreaTheme);
        builder.setMessageAreaTheme(messageAreaTheme);
        builder.setToolbarAreaTheme(toolbarAreaTheme);
        builder.setSystemAlertsTheme(systemAlertsTheme);
        builder.setPreChatTheme(preChatTheme);
        builder.setNotificationsTheme(notificationsTheme);

        return builder.build();
    }
}

File: app/src/main/java/com/helpcrunch/demo/design/CustomTheme.kt
package com.helpcrunch.demo.design

import android.content.Context
import androidx.annotation.ColorRes
import androidx.core.content.ContextCompat
import com.helpcrunch.library.R
import com.helpcrunch.library.core.options.theme.HCAvatarTheme
import com.helpcrunch.library.core.options.theme.HCChatAreaTheme
import com.helpcrunch.library.core.options.theme.HCMessageAreaTheme
import com.helpcrunch.library.core.options.theme.HCNotificationsTheme
import com.helpcrunch.library.core.options.theme.HCPreChatTheme
import com.helpcrunch.library.core.options.theme.HCSystemAlertsTheme
import com.helpcrunch.library.core.options.theme.HCTheme
import com.helpcrunch.library.core.options.theme.HCToolbarAreaTheme

class CustomTheme {
    companion object {
        fun create(ctx: Context): HCTheme {
            return HCTheme.build {
                val avatarTheme = HCAvatarTheme.build {
                    setPlaceholderBackgroundColor(ctx.color(R.color.hc_color_chat_bubble_dark_other))
                    setPlaceholderTextColor(ctx.color(R.color.hc_color_white_dark))
                }

                setChatAreaTheme(HCChatAreaTheme.build {
                    setBackgroundColor(ctx.color(R.color.hc_color_chat_dark_bg))
                    setAdditionalMessagesBackgroundColor(ctx.color(R.color.hc_color_chat_bubble_dark_other))
                    setIncomingBubbleColor(ctx.color(R.color.hc_color_chat_bubble_dark_other))
                    setOutcomingBubbleColor(ctx.color(R.color.hc_color_chat_bubble_dark_me))
                    setIncomingBubbleTextColor(ctx.color(R.color.hc_color_white_dark))
                    setOutcomingBubbleTextColor(ctx.color(R.color.hc_color_white_dark))
                    setIncomingBubbleLinkColor(ctx.color(R.color.hc_color_white_dark))
                    setOutcomingBubbleLinkColor(ctx.color(R.color.hc_color_white_dark))
                    setSystemMessageColor(ctx.color(R.color.hc_color_white_dark))
                    setTimeTextColor(ctx.color(R.color.hc_color_white_op40))
                    setProgressViewsColor(ctx.color(R.color.hc_color_white_dark))
                    setFabDownBackgroundColor(ctx.color(R.color.hc_color_toolbar_dark_bg))
                    setBrandingType(HCChatAreaTheme.Branding.DARK)
                    setAvatarTheme(avatarTheme)
                    setIncomingCodeBackgroundColor(ctx.color(R.color.hc_color_code_bg_incoming_dark))
                    setOutcomingCodeBackgroundColor(ctx.color(R.color.hc_color_code_bg_outcoming_dark))
                    setIncomingCodeTextColor(ctx.color(R.color.hc_color_code_color))
                    setOutcomingCodeTextColor(ctx.color(R.color.hc_color_code_color))
                    setIncomingBlockQuoteColor(ctx.color(R.color.hc_color_block_quote_incoming_dark))
                    setOutcomingBlockQuoteColor(ctx.color(R.color.hc_color_block_quote_outcoming_dark))
                    // File
                    setIncomingFileTextColor(ctx.color(R.color.hc_color_white))
                    setOutcomingFileTextColor(ctx.color(R.color.hc_color_white))
                    setOutcomingFileBackgroundColor(ctx.color(R.color.hc_color_code_color))
                    setIncomingFileBackgroundColor(ctx.color(R.color.hc_color_code_color))
                    setOutcomingFileIconColor(ctx.color(R.color.hc_color_code_color))
                    setIncomingFileIconColor(ctx.color(R.color.hc_color_white))
                    setAttachmentIconsColor(ctx.color(R.color.hc_color_chat_bubble_dark_other))
                    setAuthorNameColor(ctx.color(R.color.hc_color_white_dark))
                })

                setMessageAreaTheme(HCMessageAreaTheme.build {
                    setButtonType(HCMessageAreaTheme.ButtonType.TEXT)
                    setInputFieldTextColor(ctx.color(R.color.hc_color_chats_text_dark))
                    setInputFieldTextHintColor(ctx.color(R.color.hc_color_chats_text_hint_dark))
                    setInputOutlineColor(ctx.color(R.color.hc_color_toolbar_dark_bg))
                    setBackgroundColor(ctx.color(R.color.hc_color_message_bar_dark_bg))
                    setMessageMenuBackgroundColor(ctx.color(R.color.hc_color_chat_dark_bg))
                    setMessageMenuSummaryTextColor(ctx.color(R.color.hc_color_menu_summary_color_dark))
                    setMessageMenuIconColor(ctx.color(R.color.hc_color_menu_summary_color_dark))
                    setMessageMenuTextColor(ctx.color(R.color.hc_color_white_dark))
                })

                setToolbarAreaTheme(HCToolbarAreaTheme.build {
                    setBackgroundColor(ctx.color(R.color.hc_color_toolbar_dark_bg))
                    setAgentsTextColor(ctx.color(R.color.hc_color_white_dark))
                    setStatusBarColor(ctx.color(R.color.hc_color_toolbar_dark_bg))
                    setOutlineColor(ctx.color(R.color.hc_color_chat_dark_bg))
                    setAvatarTheme(avatarTheme)
                })

                setSystemAlertsTheme(HCSystemAlertsTheme.build {
                    setToastsBackgroundColor(ctx.color(R.color.hc_color_white_op90))
                    setToastsTextColor(ctx.color(R.color.hc_main_dark))
                    setDialogsHeaderColor(ctx.color(R.color.hc_color_toolbar_dark_bg))
                    setWelcomeMessageBackgroundColor(ctx.color(R.color.hc_color_toolbar_dark_bg))
                    setWelcomeMessageTextColor(ctx.color(R.color.hc_color_white_dark))
                    setWarningDialogsHeaderColor(ctx.color(R.color.hc_color_bg_button_enabled_dark))
                    setDialogAcceptButtonTextColor(ctx.color(R.color.hc_color_white_dark))
                    setDialogCancelButtonTextColor(ctx.color(R.color.hc_color_white_dark))
                    setDialogBackgroundColor(ctx.color(R.color.hc_color_chat_dark_bg))
                    setDialogMessageTextColor(ctx.color(R.color.hc_color_white_dark))
                })

                setPreChatTheme(HCPreChatTheme.build {
                    setButtonContinueBackgroundSelector(R.drawable.btn_hc_rounded_dark)
                    setInputFieldTextColor(ctx.color(R.color.hc_color_chats_text_dark))
                    setInputFieldTextHintColor(ctx.color(R.color.hc_color_chats_text_hint_dark))
                    setInputFieldBackgroundDrawableRes(R.drawable.bg_hc_chat_field_dark)
                    setBackgroundColor(ctx.color(R.color.hc_color_chat_dark_bg))
                    setMessageBackgroundColor(ctx.color(R.color.hc_color_toolbar_dark_bg))
                    setMessageTextColor(ctx.color(R.color.hc_color_white_dark))
                })

                setNotificationsTheme(HCNotificationsTheme.build {
                    setAvatarTheme(avatarTheme)
                    setColor(ctx.color(R.color.hc_color_chat_bubble_dark_other))
                })
            }
        }
    }
}

fun Context.color(@ColorRes colorRes: Int): Int {
    return ContextCompat.getColor(this, colorRes)
}
